keypair/keypair.go

```diff
@@ -0,0 +1,124 @@
+package keypair
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/mitchellh/go-homedir"
+	"github.com/pkg/errors"
+)
+
+type Config struct {
+	PrivateKey *rsa.PrivateKey
+	PublicKey  *rsa.PublicKey
+	KeyPrefix  string
+	KeyPath    string
+}
+
+func (c *Config) GetPrivateKeyPath() string {
+	return fmt.Sprintf("%s/%s_private.pem", c.KeyPath, c.KeyPrefix)
+}
+
+func (c *Config) GetPublicKeyPath() string {
+	return fmt.Sprintf("%s/%s_public.pem", c.KeyPath, c.KeyPrefix)
+}
+
+func ParsePrivateKey(privateKeyPath string) (*rsa.PrivateKey, error) {
+	expandedPrivateKeyPath, err := homedir.Expand(privateKeyPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "Invalid Path to private key")
+	}
+
+	privKeyBytes, err := ioutil.ReadFile(expandedPrivateKeyPath)
+	if err != nil {
+		return nil, errors.Errorf("Unable to read private key file")
+	}
+
+	privPemBlock, _ := pem.Decode(privKeyBytes)
```

Comment:
that _ an error?

Reply:
Actually not! It returns (p *Block, rest []byte)
https://pkg.go.dev/encoding/pem#Decode
I only needed the privPemBlock variable to do stuff, so I made the rest _